packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -230,15 +230,30 @@ class FlutterDevice {
       isWaitingForVm = true;
       vm_service.VmService service;
       if (!disableDds) {
```

Comment:
The tool integration tests could be written to cover this partially - i.e. create an app template that crashes immediately and connect to it. This should work with the flutter tester, right?